compiler/src/dotty/tools/dotc/transform/FirstTransform.scala

```diff
@@ -144,7 +144,13 @@ class FirstTransform extends MiniPhase with InfoTransformer { thisPhase =>
   }
 
   override def transformIdent(tree: Ident)(implicit ctx: Context): Tree =
-    if (tree.isType) toTypeTree(tree) else constToLiteral(tree)
+    if (tree.isType) {
+      toTypeTree(tree)
+    } else if (tree.name != nme.WILDCARD) {
```

Comment:
Can't we check ctx.mode.is(Pattern) here too ?

Reply:
We don't want to, since we actually want to inline non-wildcard idents. One example that compiling the stdlib thankfully caught is `@switch` annotation, which needs to see appropriate inlined literals. I'll add a comment explaining as much.